src/tdigest.rs

```diff
@@ -0,0 +1,269 @@
+// fn scale(q: f32, delta: f32) -> f32 {
+//     if q >= 0.5 {
+//         delta - delta * (2.0 - 2.0 * q).sqrt() / 2.0
+//     } else {
+//         delta * (2.0 * q).sqrt() / 2.0
+//     }
+// }
+
+// inverse of the scale function (commented above)
+fn inverse_scale(k: f32, delta: f32) -> f32 {
+    let kd_ratio = k / delta; // funny gamer joke. see the reason it's funny is because
+    if kd_ratio >= 0.5 {
+        1.0 - 2.0 * f32::powi(1.0 - kd_ratio, 2)
+    } else {
+        2.0 * f32::powi(kd_ratio, 2)
+    }
+}
+
+#[derive(Clone, Copy)]
+struct Centroid {
+    mean: f32,
+    weight: i32,
+}
+
+impl Centroid {
+    fn new(mean: f32, weight: i32) -> Self {
+        Self {
+            mean: mean,
+            weight: weight,
+        }
```

Comment:
Rust tip: whenever the variable has the same name of the struct field, you can use the equivalent shorthand initialization:
```rust
Self { mean, weight }
```